### PR TITLE
doc/dev: resurrect hidden dev/cephadm.rst

### PR DESCRIPTION
**Pull request by sebastian-philipp**
some time ago, a cephadm folder was added that did hide the
old dev documentation page.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by varshar16 — APPROVED** [review]
Looks Good. It would be nice if you can update the index too.
From `Cephadm Feature Planning` ->  `Cephadm Developer Guide`
https://github.com/ceph/ceph/blob/master/doc/cephadm/index.rst

**Comment by sebastian-philipp** [comment]
imo we should not link the dev guide form the user documentation. The user docs are already convoluted and hard to follow. Adding the dev guide is not helpful.
